### ISSUE TITLE
`celery multi start` raises when used with preload options

### ISSUE THREAD

**Issue by maconfr** [celery/celery#7683]
Preload options do not work with `celery multi start` and `celery report`.

This can easily reproduced by the following steps.

Setup a test project like this:

````bash
$ cd /tmp/
$ python3 -m venv celery_venv
$ . celery_venv/bin/activate
$ git clone https://github.com/celery/celery.git
Cloning into 'celery'...
[...]
$ cd celery
$ pip install -e .
[...]
$ celery report

software -> celery:5.3.0b1 (dawn-chorus) kombu:5.3.0b1 py:3.9.2
            billiard:4.0.2 py-amqp:5.1.1
platform -> system:Linux arch:64bit, ELF
            kernel version:5.10.0-16-amd64 imp:CPython
loader   -> celery.loaders.default.Loader
settings -> transport:amqp results:disabled

deprecated_settings: None

````

Then create a `proj.py`. You can simply use the [Preload options](https://docs.celeryproject.org/en/stable/userguide/extending.html#preload-options)
example.

Starting a worker with the `--template` option works with `celery worker`:

````bash
$ celery -A proj worker --template foo
[2022-08-04 12:57:19,836: WARNING/MainProcess] No hostname was supplied. Reverting to default 'localhost'
 
 -------------- celery@d6a4 v5.3.0b1 (dawn-chorus)
--- ***** ----- 
-- ******* ---- Linux-5.10.0-16-amd64-x86_64-with-glibc2.31 2022-08-04 12:57:19
- *** --- * --- 
- ** ---------- [config]
- ** ---------- .> app:         __main__:0x7fcbd6c5a1c0
- ** ---------- .> transport:   amqp://guest:**@localhost:5672//
- ** ---------- .> results:     disabled://
- *** --- * --- .> concurrency: 8 (prefork)
-- ******* ---- .> task events: OFF (enable -E to monitor tasks in this worker)
--- ***** ----- 
 -------------- [queues]
                .> celery           exchange=celery(direct) key=celery
[...]
````

But starting a worker with the `--template` option fails with `celery multi start`:

````bash
$ celery -A proj multi start worker1 --template foo
Traceback (most recent call last):
  File "/tmp/celery_venv/bin/celery", line 33, in <module>
    sys.exit(load_entry_point('celery', 'console_scripts', 'celery')())
  File "/tmp/celery/celery/__main__.py", line 15, in main
    sys.exit(_main())
  File "/tmp/celery/celery/bin/celery.py", line 217, in main
    return celery(auto_envvar_prefix="CELERY")
  File "/tmp/celery_venv/lib/python3.9/site-packages/click/core.py", line 1130, in __call__
    return self.main(*args, **kwargs)
  File "/tmp/celery_venv/lib/python3.9/site-packages/click/core.py", line 1055, in main
    rv = self.invoke(ctx)
  File "/tmp/celery_venv/lib/python3.9/site-packages/click/core.py", line 1657, in invoke
    return _process_result(sub_ctx.command.invoke(sub_ctx))
  File "/tmp/celery_venv/lib/python3.9/site-packages/click/core.py", line 1404, in invoke
    return ctx.invoke(self.callback, **ctx.params)
  File "/tmp/celery_venv/lib/python3.9/site-packages/click/core.py", line 760, in invoke
    return __callback(*args, **kwargs)
  File "/tmp/celery_venv/lib/python3.9/site-packages/click/decorators.py", line 26, in new_func
    return f(get_current_context(), *args, **kwargs)
  File "/tmp/celery/celery/bin/base.py", line 134, in caller
    return f(ctx, *args, **kwargs)
TypeError: multi() got an unexpected keyword argument 'template'
````

The `celery report` command has the same issue, which is quite annoying when the custom preload option is required instead of optional.

````bash
$ celery -A proj report --template foo
Traceback (most recent call last):
  File "/tmp/celery_venv/bin/celery", line 33, in <module>
    sys.exit(load_entry_point('celery', 'console_scripts', 'celery')())
  File "/tmp/celery/celery/__main__.py", line 15, in main
    sys.exit(_main())
  File "/tmp/celery/celery/bin/celery.py", line 217, in main
    return celery(auto_envvar_prefix="CELERY")
  File "/tmp/celery_venv/lib/python3.9/site-packages/click/core.py", line 1130, in __call__
    return self.main(*args, **kwargs)
  File "/tmp/celery_venv/lib/python3.9/site-packages/click/core.py", line 1055, in main
    rv = self.invoke(ctx)
  File "/tmp/celery_venv/lib/python3.9/site-packages/click/core.py", line 1657, in invoke
    return _process_result(sub_ctx.command.invoke(sub_ctx))
  File "/tmp/celery_venv/lib/python3.9/site-packages/click/core.py", line 1404, in invoke
    return ctx.invoke(self.callback, **ctx.params)
  File "/tmp/celery_venv/lib/python3.9/site-packages/click/core.py", line 760, in invoke
    return __callback(*args, **kwargs)
  File "/tmp/celery_venv/lib/python3.9/site-packages/click/decorators.py", line 26, in new_func
    return f(get_current_context(), *args, **kwargs)
TypeError: report() got an unexpected keyword argument 'template'
````

Adjusting the function arguments as suggested by JurgenFiGO might be a solution.

An other approach might be to delete each user option from the `kwargs` variable in the `handle_preload_options` decorator.
But I am not sure if this will break other use cases. So, adjusting parameter list of all functions that are using this decorator seems to be the safest way.  
The decorator is calling the decorated this way: `f(ctx, *args, **kwargs)`  
See `handle_preload_options` decorator in `celery/bin/base.py` line 119.

_Originally posted by @maconfr in https://github.com/celery/celery/discussions/7136#discussioncomment-1900400_

**Comment by KoleS46** [comment]
Hello all :wave:,
I have also encountered this issue but with the `shell` command.
And before searching for the issue here I did some digging on my own and spotted that the `shell` [command](https://github.com/celery/celery/blob/main/celery/bin/shell.py#L118) as the other erring commands [report](https://github.com/celery/celery/blob/main/celery/bin/celery.py#L176), [multi](https://github.com/celery/celery/blob/main/celery/bin/multi.py#L472) don't have the `**kwargs` argument added like the `worker` [command](https://github.com/celery/celery/blob/main/celery/bin/worker.py#L300) has.

So basically:
```python
def shell(
    ctx, ipython=False, bpython=False, python=False, without_tasks=False, eventlet=False,
    gevent=False, **kwargs
):
```

**Comment by auvipy** [comment]
thanks, I am going to fix this here https://github.com/celery/celery/pull/7994

**Comment by Nusnus** [comment]
Moving to Milestone 5.3.x for now to reduce the length for 5.3.**0** release.
